Marlin/cardreader.cpp

```diff
@@ -611,7 +652,11 @@ void CardReader::chdir(const char * relpath) {
 
   if (workDir.isOpen()) parent = &workDir;
 
-  if (!newDir.open(*parent, relpath, O_READ)) {
+  #ifdef USE_NEW_SD_FAT_LIB
+    if (!newDir.open(parent, relpath, O_READ)) {
+  #else
+    if (!newDir.open(*parent, relpath, O_READ)) {
+  #endif
```

Comment:
Better:
```cpp
if (!newDir.open(
  #if ENABLED(USE_NEW_SD_FAT_LIB)
    parent, relpath, O_READ
  #else
    *parent, relpath, O_READ
  #endif
)) {
```
You get the idea. Note the `ENABLED` macro. This is required to allow for `Makefile` building.

Reply:
Thank you very much for taking time to give this feedback. I'll update and make a new PR.